llvm/test/CodeGen/NVPTX/compute-ptx-value-vts.ll

```diff
@@ -0,0 +1,61 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc < %s -march=nvptx64 -mcpu=sm_20 | FileCheck %s
+
+target triple = "nvptx-nvidia-cuda"
+
+define <6 x half> @half6() {
```

Comment:
We do want to have a more exhaustive coverage here for non-power-of-2 sized vectors.

For the sake of completeness I'd enumerate all variants for i1/i8/i16/i32/i64 vectors with the total size <= 24 bytes (16 bytes to cover the max possible vector size we may ever lower natively + one element, so we cover cases of power-of-2 vector+remainder). Vectorized lowering of loads/stores is pretty hairy, so it would be nice to know that we're not missing any other uncommon corner cases.

Reply:
I used the below script to write the tests. Let me know if I misunderstood the ask,

```
#!/usr/bin/python3

import math

MAX_BYTES = 24
MAX_ELEMENT_COUNT = 200

pairs = []

for data_size in [1, 8, 16, 32, 64]:
    for element_count in range(1, MAX_ELEMENT_COUNT):
        if element_count!= 1 and math.log2(element_count).is_integer():
            continue
        if element_count * data_size > MAX_BYTES * 8:
            continue

        pairs.append((data_size, element_count))

pairs.reverse()
        
for data_size, element_count in pairs:
    if data_size == 1:
        data_name = "bit"
    elif data_size == 8:
        data_name = "byte"
    elif data_size == 16:
        data_name = "short"
    elif data_size == 32:
        data_name = "int"
    elif data_size == 64:
        data_name = "long"
    else:
        assert(False)
    
    print(f"define <{element_count} x i{data_size}> @{data_name}{element_count}()" " {\n" f"\tret <{element_count} x i{data_size}> zeroinitializer\n" "}\n")
```

Reply:
What I, roughly, had in mind is something like this:

```
interesting_cases = set()
for data_size in [1, 8, 16, 32, 64]:
   for pow2 in range(1,5):
      vector_size = element_count * data_size 
      if vector_size > MAX_BYTES:
          break;
      interesting_cases.add([data_size, element_count-1])
      interesting_cases.add([data_size, element_count])
      interesting_cases.add([data_size, element_count+1])
```

Reply:
Ahh, I see. Thanks for the clarification. 

I've added this in `vector-returns.ll` because the test cases won't touch the `ComputePTXValueVTs()` bug fix.